src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    [CompilerTrait(CompilerFeature.ReadOnlyReferences)]
+    public class RecordTests : CompilingTestBase
+    {
+        [Fact]
+        public void RecordLanguageVersion()
```

Comment:
>RecordLanguageVersion [](start = 20, length = 21)

Also try `class Point;`.  That should not be permitted in older language versions either.